packages/file-search/src/browser/quick-file-open-contribution.ts

```diff
@@ -47,7 +48,8 @@ export class QuickFileOpenFrontendContribution implements CommandContribution, K
     registerKeybindings(keybindings: KeybindingRegistry): void {
         keybindings.registerKeybinding({
             command: quickFileOpen.id,
-            keybinding: 'ctrlcmd+p'
+            keybinding: 'ctrlcmd+p',
+            context: WorkspaceKeybindingContexts.workspaceOpened
```

Comment:
we should use `when` instead, `context` is deprecated and will be removed

Reply:
There is a new when context in VS Code: https://github.com/microsoft/vscode-docs/commit/3ca603d30295db5728d6f403bfd7b0f405264737

We should implement it in https://github.com/eclipse-theia/theia/blob/f4a92a7c8fd15140f542cca55d2d67441158a819/packages/workspace/src/browser/workspace-frontend-contribution.ts#L57-L66

and then use as `when: 'workspaceState != empty'`

Reply:
> we should use `when` instead, `context` is deprecated and will be removed

Thank you for the suggestion and guidance, I updated the code to instead implement and use the `workspaceState` when clause.